assets.go

```diff
@@ -1,354 +1,134 @@
-	Position *Point
-	Scale    *Point
-	Anchor   *Point
-	Rotation float32
-	Color    color.Color
-	Alpha    float32
-	Region   *Region
-}
-
-func NewSprite(region *Region, x, y float32) *Sprite {
-	return &Sprite{
-		Position: &Point{x, y},
-		Scale:    &Point{1, 1},
-		Anchor:   &Point{0, 0},
-		Rotation: 0,
-		Color:    color.White,
-		Alpha:    1,
-		Region:   region,
+// Resource returns the given resource, and an error if it didn't succeed
+func (formats *Formats) Resource(url string) (Resource, error) {
+	ext := filepath.Ext(url)
+	if loader, ok := Files.formats[ext]; ok {
+		return loader.Resource(url)
 	}
+	return nil, ResourceLoaderNotFoundError{Format: ext, URL: url}
 }
 
-func ImageToNRGBA(img image.Image, width, height int) *image.NRGBA {
-	newm := image.NewNRGBA(image.Rect(0, 0, width, height))
-	draw.Draw(newm, newm.Bounds(), img, image.Point{0, 0}, draw.Src)
-
-	return newm
+// A ResourceLoaderNotFoundError is returned whenever the specified file format has no registered `FileLoader` to
+// return the requested resource.
+type ResourceLoaderNotFoundError struct {
+	Format string
+	URL    string
 }
 
-// ImageObject is a pure Go implementation of a `Drawable`
-type ImageObject struct {
-	data *image.NRGBA
+func (r ResourceLoaderNotFoundError) Error() string {
+	return fmt.Sprintf("no `FileLoader` registered for file format %s (within url: %s)", r.Format, r.URL)
 }
 
-// NewImageObject creates a new ImageObject given the image.NRGBA reference
-func NewImageObject(img *image.NRGBA) *ImageObject {
-	return &ImageObject{img}
+// A ResourceNotLoadedError is returned whenever the requested `Resource` was unable to be located within the memory.
+// This usually indicates the `Load`-step failed.
+type ResourceNotLoadedError struct {
+	URL string
 }
 
-// Data returns the entire image.NRGBA object
-func (i *ImageObject) Data() interface{} {
-	return i.data
+func (r ResourceNotLoadedError) Error() string {
+	return fmt.Sprintf("the `FileLoader` was unable to find `Resource` in memory: %s", r.URL)
 }
 
-// Width returns the maximum X coordinate of the image
-func (i *ImageObject) Width() int {
-	return i.data.Rect.Max.X
+// ResourceOpenError is returned whenever the assets manager was unable to access the file. It was therefore also
+// unable to send the file to the `FileLoader`.
+//
+// Possible reasons for this may be that your program does not have the right permissions to open the file,
+// that the path (url) is false, or that the file simply does not exist.
+type ResourceOpenError struct {
+	URL string
+
+	// Err is the internal error of the `io` method used. Which method this is, is OS-dependant (`io` for desktop,
+	// `asset` for mobile, etc.)
+	Err error
 }
 
-// Height returns the maximum Y coordinate of the image
-func (i *ImageObject) Height() int {
-	return i.data.Rect.Max.Y
+func (r ResourceOpenError) Error() string {
+	return fmt.Sprintf("the assets manager was unable to open `Resource`: %s (%s)", r.URL, r.Err.Error())
```

Comment:
The `fmt` package is aware of the `error` interface, and much in the same way as you don't need to call `String` on `fmt.Stringer` values, you don't need to call `Error` on values implementing the `error` interface. It is good to know that values implementing both the `fmt.Stringer` and `error` interface, then `fmt` will invoke `Error` as the `error` interface takes precedence.

```diff
r.Err.Error()
r.Err
```